pkg/resources/task_acceptance_test.go

```diff
@@ -1,564 +1,970 @@
+		WithSchedule(schedule)
+	secondTaskStandaloneModelDisabled := model.TaskWithId("second_task", finalizerTaskId, false, statement).
+		WithSchedule(schedule)
+	secondTaskStandaloneModelDisabled.SetDependsOn([]string{firstTaskStandaloneModelDisabled.ResourceReference()})
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Task),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModel(t, firstTaskStandaloneModel) + config.FromModel(t, secondTaskStandaloneModel),
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, firstTaskStandaloneModel.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanTrue).
+						HasSuspendTaskAfterNumFailuresString("1"),
+					resourceshowoutputassert.TaskShowOutput(t, firstTaskStandaloneModel.ResourceReference()).
+						HasSchedule(schedule).
+						HasState(sdk.TaskStateStarted),
+					resourceassert.TaskResource(t, secondTaskStandaloneModel.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanTrue),
+					resourceshowoutputassert.TaskShowOutput(t, secondTaskStandaloneModel.ResourceReference()).
+						HasSchedule(schedule).
+						HasState(sdk.TaskStateStarted),
+				),
+			},
+			// Change the second task to run after the first one (creating a DAG)
+			{
+				Config: config.FromModel(t, rootTaskModel) + config.FromModel(t, childTaskModel),
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, rootTaskModel.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanTrue).
+						HasSuspendTaskAfterNumFailuresString("2"),
+					resourceshowoutputassert.TaskShowOutput(t, rootTaskModel.ResourceReference()).
+						HasSchedule(schedule).
+						// HasTaskRelations(sdk.TaskRelations{FinalizerTask: &finalizerTaskId}).
+						HasState(sdk.TaskStateStarted),
+					resourceassert.TaskResource(t, childTaskModel.ResourceReference()).
+						HasEnabledString(r.BooleanTrue),
+					resourceshowoutputassert.TaskShowOutput(t, childTaskModel.ResourceReference()).
+						// HasTaskRelations(sdk.TaskRelations{FinalizedRootTask: &rootTaskId}).
+						HasState(sdk.TaskStateStarted),
+				),
+			},
+			// Change tasks in DAG to standalone tasks (disabled to check if resuming/suspending works correctly)
+			{
+				Config: config.FromModel(t, firstTaskStandaloneModelDisabled) + config.FromModel(t, secondTaskStandaloneModelDisabled),
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, firstTaskStandaloneModelDisabled.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanFalse).
+						HasSuspendTaskAfterNumFailuresString("10"),
```

Comment:
We should also check that finalized/finalizer fields are empty.

Reply:
Yup, added.